File: server/src/knowledge/ingestors/excelIngestor.js
import { BaseIngestor } from '../baseIngestor.js';

// Needs a real xlsx parser (e.g. `xlsx` / `exceljs`) — not installed yet.
export class ExcelIngestor extends BaseIngestor {
  constructor() {
    super('document');
  }

  async ingest(_source) {
    throw new Error('ExcelIngestor not implemented — add `xlsx` package and sheet-to-row chunking');
  }
}

File: server/app/knowledge/ingestors/excel_ingestor.py
from ..base_ingestor import BaseIngestor


class ExcelIngestor(BaseIngestor):
    def __init__(self):
        super().__init__("document")

    async def ingest(self, source: dict) -> list[dict]:
        raise NotImplementedError("ExcelIngestor not implemented — add openpyxl and sheet-to-row chunking")
